File: neetcode/duplicate_integer.py

from typing import List

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

li: List[int] = [1, 2, 3, 3, 4]
obj = Solution()
print(obj.hasDuplicate(li))

File: neetcode/HashmapsAndArrays/duplicate_integer.cpp
#include <exception>
#include<vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
  bool hasDuplicate(vector<int>& nums){
    unordered_set<int> set;
    for (int i : nums)
    { 
      try
      {
        auto result = set.insert(i);
        bool condition = result.second;
        if(condition)
          continue;
        else {
          return true;
        }  
        
      }
      catch (exception e)
      {
          return false;
      }
    }
    return false;
  }
};
